src/main/scala/firrtl/Emitter.scala

```diff
@@ -146,6 +149,8 @@ object EmitAllModulesAnnotation extends HasShellOptions {
               RunFirrtlTransformAnnotation(new SystemVerilogEmitter),
               EmitAllModulesAnnotation(classOf[SystemVerilogEmitter])
             )
+          case "experimental-rtlil" | "rtlil" =>
+            Seq(RunFirrtlTransformAnnotation(Dependency(RtlilEmitter)), EmitCircuitAnnotation(RtlilEmitter.getClass))
```

Comment:
Is your backend capable of emitting one file per module? If not, you shouldn't add it here.
The `smt2` backend for example always produces a single file which is why it is only supported by the `-E`  option.

Reply:
I just saw that you support `EmitAllModulesAnnotation` in your emitter code. You probably just forgot to add it here.